src/main/java/de/vogel612/testclient_javabot/client/ClientGui.java

```diff
@@ -1,32 +1,41 @@
 package de.vogel612.testclient_javabot.client;
 
 import java.util.concurrent.CountDownLatch;
-import java.util.concurrent.Executors;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
+import javafx.application.Application;
 import javafx.application.Platform;
-import javafx.embed.swing.JFXPanel;
 import javafx.fxml.FXMLLoader;
 import javafx.scene.Scene;
 import javafx.scene.layout.BorderPane;
-
-import javax.swing.JFrame;
+import javafx.stage.Stage;
 
 import com.gmail.inverseconduit.chat.ChatWorker;
 import com.gmail.inverseconduit.datatype.ChatMessage;
 
 import de.vogel612.testclient_javabot.client.controller.ChatRenderController;
 
-public class ClientGui implements ChatWorker {
+public class ClientGui extends Application implements ChatWorker {
 
 	private static final Logger LOGGER = Logger.getLogger(ClientGui.class.getName());
 
 	private ChatRenderController controller;
-	private JFrame frame;
-	final CountDownLatch latch = new CountDownLatch(1);
+	private Stage stage;
+	private static ClientGui instance;
+	public static final CountDownLatch latch = new CountDownLatch(1);
+	
+	public static ClientGui getInstance() {
+		return instance;
+	}
 
-	public void initFX(final JFXPanel fxPanel) throws Exception {
+	public ClientGui() {
```

Comment:
Well, it is required. Because we are loading the configuration and later showing the GUI, which needs the instance of ClientGui so that it can show the Stage(later). If there is any other way to retain the instance of ClientGui which is created when static `launch()` is called from TestProgram constructor, I will definitely like to know.